Composer/packages/extensions/intellisense/src/components/IntellisenseTextField.tsx

```diff
@@ -0,0 +1,170 @@
+      }
+    };
+
+    document.body.addEventListener('click', outsideClickHandler);
+    document.body.addEventListener('keyup', keyupHandler);
+
+    return () => {
+      document.body.removeEventListener('click', outsideClickHandler);
+      document.body.removeEventListener('keyup', keyupHandler);
+    };
+  }, []);
+
+  // When textField value is changed
+  const onValueChanged = (newValue: string) => {
+    setTextFieldValue(newValue);
+
+    onChange(newValue);
+  };
+
+  // Handles selection of completion items and validation through keyboard (Up Down to navigate and Enter to validate)
+  const onKeyUpMainComponent = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    switch (event.key) {
+      case 'ArrowDown':
+        completionItems?.length && setSelectedCompletionItem((index) => (index + 1) % completionItems?.length);
+        break;
+      case 'ArrowUp':
+        completionItems?.length &&
+          setSelectedCompletionItem((index) => (completionItems?.length + index - 1) % completionItems?.length);
+        break;
+      case 'Enter':
+        setValueToSelectedCompletionItem(selectedCompletionItem);
+        break;
+    }
+  };
+
+  // Handles validation of a suggested completion item when clicking on it
+  const onClickCompletionItem = (suggestionIndex: number) => {
+    setValueToSelectedCompletionItem(suggestionIndex);
+  };
+
+  // Prevents cursor from moving around in textField when going through the list of completion items
+  const onKeyDownTextField = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    switch (event.key) {
+      case 'ArrowDown':
+      case 'ArrowUp':
+        if (completionItems?.length) {
+          event.preventDefault();
+        }
+        break;
+    }
+    // Typing also modifies the cursor position
+    setCursorPosition((event.target as HTMLInputElement).selectionStart || 0);
+  };
+
+  // Updates position of cursor
+  const onClickTextField = (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
+    setCursorPosition((event.target as HTMLInputElement).selectionStart || 0);
+  };
+
+  // Set textField value to completion item value
+  const setValueToSelectedCompletionItem = (index: number) => {
+    if (completionItems.length >= index) {
+      const selectedSuggestion = completionItems[index].insertText || '';
+      const range = completionItems[index].data.range;
+
+      if (range) {
+        const newValue =
+          textFieldValue.substr(0, range.start.character) +
+          selectedSuggestion +
+          textFieldValue.substr(range.end.character);
+        onValueChanged(newValue);
+      } else {
+        onValueChanged(selectedSuggestion);
+      }
+
+      // This makes sure we do not show the completion items after a value is picked from the list
+      didComplete.current = true;
+      setShowCompletionList(false);
+    }
+  };
+
+  return (
+    <div onKeyUp={onKeyUpMainComponent} ref={mainContainerRef} style={{ position: 'relative' }}>
```

Comment:
Consider if inline styles are required here or if you can use emotion/styled-components.  You may want to provide the className in the props to allow callers to override styles.